src/GafferScene/SceneAlgo.cpp

```diff
@@ -127,6 +127,22 @@ struct ThreadablePathAccumulator
 
 };
 
+struct ThreadablePathHashAccumulator
+{
+	ThreadablePathHashAccumulator(): m_h1Accum( 0 ), m_h2Accum( 0 ){}
+
+	bool operator()( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
+	{
+		IECore::MurmurHash h;
+		h.append( &path.front(), path.size() );
```

Comment:
I experimented with hashing the string address here rather than the contents, but didn't see any improvement : https://github.com/johnhaddon/gaffer/commit/bd4356db3afbe9ab5e01c0da182740c8ecaf995d.